spec.md

```diff
@@ -92,39 +91,52 @@ The `Span` interface must have the following capabilities:
+- Start a new `Span`. There must be a way for the caller to specify a parent `Span`, an explicit start timestamp (other than "now"), and an initial set of `Span` tags. **(py: `start_span`, go: `StartSpanWithOptions`)**
+- Return an `Injector` instance given a format identifier, or nil/null if the format is not supported by the implementation (see [Injector and Extractor](#injector-and-extractor))
+- Return an `Extractor` instance given a format identifier, or nil/null if the format is not supported by the implementation (see [Injector and Extractor](#injector-and-extractor))
 
-### SpanPropagator
+A note about **"format identifiers"** per the `Injector` and `Extractor` lookup methods above: the nature of the "format identifier" may vary from platform to platform, but in all cases they should be drawn from a global namespace. New formats must not *require* changes to the core OpenTracing platform APIs, though those core platform APIs must define a few basic/general formats (like string maps and binary blobs). For example, if the maintainer of EsotericRPCFramework wanted to define an EsotericRPCFramework injection and extraction format, she or he must be able to do so without sending a PR to OpenTracing maintainers (though of course OpenTracing implementations are not required to suppor the EsotericRPCFramework format).
 
-A SpanPropagator is responsible (a) for encoding Span instances in a manner suitable for propagation, and (b) for taking that encoded data and using it
-to generate Span instances that are placed appropriately in the overarching Trace. Typically the propagation will take place across an RPC boundary, but message queues and other IPC mechanisms are also reasonable places to use a SpanPropagator.
+## Injector and Extractor
 
-The encoded form of a propagated span is divided into two components:
+The `Injector` and `Extractor` interfaces allow Traces to propagate across process boundaries. They are not entirely symmetrical interfaces since `Span` injection is a fundamentally lossy process (most of the `Span`'s state is meant to be recorded out-of-band from the application's own communication path, whereas Inject-ed state is meant to be sent in-band with the application data).
 
-1. The core identifying information for the Span, referred to as the "context snapshot" (for example, in Dapper this would include a `trace_id`, a `span_id`, and a bitmask representing the sampling status for the given trace)
-1. Any trace attributes (per Span's ability to set trace attributes that propagate across process boundaries)
+- An `Injector` has one method: `InjectSpan`. InjectSpan takes two arguments: a `Span` instance and an "carrier" object in which to inject that `Span` for cross-process propagation. The type of the carrier depends on the particular type of `Injector`, and the type of the `Injector` in turn depends on the "format identifier".
```

Comment:
@marcingrzejszczak fyi this is a recent change to OT, which makes it more clear where the propagation responsibility lies. Ex the Injected state, as listed above, is only the state send over headers or similar.